### PR TITLE
Flask: improvements for running and testing

### PR DESCRIPTION
**Pull request by glasswalk3r**
Changed app to use environment variables for configuration (12 factor).
Made unit testing running on in-memory SQLite database, which makes the
tests run much faster since no temporary files are required.
The database now is initialized automatically in a idempotent way.
README updated with the last changes.

### CODE REVIEW

**Comment by amigus** [comment]
@komarserjio if you want help merging this PR with the three that I just submitted, let me know.

File: flask/README.rst [hunk 1]
```diff
@@ -48,6 +48,8 @@ Create database schema:
     $ cd YOUR_PROJECT_DIR/flask/
     $ python db.py
 
+Although you shouldn't need to do it since the application will do that in a idempotent way.
```
Comment (komarserjio) [inline flask/README.rst]: I think we can remove `Create database schema` subsection then.

File: flask/README.rst [hunk 2]
```diff
@@ -59,7 +61,25 @@ Start flask web server:
...
+.. code-block:: bash
+
+    $ export ENVIRONMENT=development
+    $ python runserver.py
+
+Or in a single line:
+
+.. code-block:: bash
+
+    $ ENVIRONMENT=development python runserver.py
```
Comment (komarserjio) [inline flask/README.rst]: Let's keep it simple and leave only this single line version.

File: flask/README.rst [hunk 3]
```diff
@@ -59,7 +61,25 @@ Start flask web server:
...
+
+For production (running in a "real" database), use the value "production" and be sure to also
```
Comment (komarserjio) [inline flask/README.rst]: Nobody should ever use this demo app in a production environment :) I'd recommend removing any suggestions about the production environment and leave only development and testing.

File: flask/notejam/config.py [hunk 4]
```diff
@@ -1,23 +1,30 @@
...
     DEVELOPMENT = True
     DEBUG = True
+    SQLALCHEMY_DATABASE_URI = 'sqlite:///' + os.path.join(os.getcwd(),
+                                                          'notejam.db')
 
 
 class TestingConfig(Config):
     TESTING = True
+    """
+    Tests will run WAY faster using in memory SQLITE database
```
Comment (komarserjio) [inline flask/notejam/config.py]: Great idea!

Comment (glasswalk3r) [reply flask/notejam/config.py]: Thanks!

File: flask/README.rst (hunk 1 above)
Comment (glasswalk3r) [reply flask/README.rst]: Done!

File: flask/README.rst (hunk 2 above)
Comment (glasswalk3r) [reply flask/README.rst]: Done!

File: flask/README.rst (hunk 3 above)
Comment (glasswalk3r) [reply flask/README.rst]: Done!

**Pull request merged**